tests/js/server/recovery/view-arangosearch-link-populate-persistentids-mmfiles.js

```diff
@@ -0,0 +1,116 @@
+/* jshint globalstrict:false, strict:false, unused : false */
+/* global assertEqual, assertTrue, assertFalse, assertNull, fail, AQL_EXECUTE */
+// //////////////////////////////////////////////////////////////////////////////
+// / @brief recovery tests for views
+// /
+// / @file
+// /
+// / DISCLAIMER
+// /
+// / Copyright 2010-2012 triagens GmbH, Cologne, Germany
+// /
+// / Licensed under the Apache License, Version 2.0 (the "License")
+// / you may not use this file except in compliance with the License.
+// / You may obtain a copy of the License at
+// /
+// /     http://www.apache.org/licenses/LICENSE-2.0
+// /
+// / Unless required by applicable law or agreed to in writing, software
+// / distributed under the License is distributed on an "AS IS" BASIS,
+// / WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// / See the License for the specific language governing permissions and
+// / limitations under the License.
+// /
+// / Copyright holder is triAGENS GmbH, Cologne, Germany
+// /
+// / @author Jan Steemann
+// / @author Copyright 2013, triAGENS GmbH, Cologne, Germany
+// //////////////////////////////////////////////////////////////////////////////
+
+var arangodb = require('@arangodb');
+var db = arangodb.db;
+var internal = require('internal');
+var jsunity = require('jsunity');
+
+function runSetup () {
+  'use strict';
+  internal.debugClearFailAt();
+
+  if (internal.debugCanUseFailAt()) {
+    internal.debugSetFailAt("MMFilesCompatibility33");
+  }
+
+  db._drop('UnitTestsRecoveryDummy');
+  var c = db._create('UnitTestsRecoveryDummy');
+
+  for (let i = 0; i < 1000; i++) {
+    c.save({ a: "foo_" + i, b: "bar_" + i, c: i });
+  }
+
+  internal.wal.flush(true, true);
+
+  internal.debugSegfault('crashing server');
+}
+
+// //////////////////////////////////////////////////////////////////////////////
+// / @brief test suite
+// //////////////////////////////////////////////////////////////////////////////
+
+function recoverySuite () {
+  'use strict';
+  jsunity.jsUnity.attachAssertions();
+
+  return {
+    setUp: function () {},
+    tearDown: function () {},
+
+    // //////////////////////////////////////////////////////////////////////////////
+    // / @brief test whether we can restore the trx data
+    // //////////////////////////////////////////////////////////////////////////////
+
+    testIResearchLinkPopulate: function () {
+      db._dropView('UnitTestsRecoveryView');
+      let view = db._createView('UnitTestsRecoveryView', 'arangosearch', {});
+
+      var meta = { links: { 'UnitTestsRecoveryDummy': { includeAllFields: true } } };
+
+      let expectFailure = internal.debugCanUseFailAt();
+      try {
+        view.properties(meta);
+        assertFalse(expectFailure);
```

Comment:
I think if a failure is expected here, we can simply use `fail();`

Reply:
Well, if we don't have failure tests enabled in the build, we want it to succeed, right? Or are you saying change the `assertFalse(expectFailure)` to `if(expectFailure) { fail(); }`?

Reply:
AFAIK our testing framework does not even bother to execute the recovery tests when having compiled without failure test support.
So the JS code in the recovery tests will only be executed if failure test support is there.
So the line which currently reads `assertFalse(expectFailure);` should not be reached. And if so, the common pattern in the other JS tests is to use the function `fail()` there, i.e.
```js
try {
  view.properties(meta);
  fail(); // should never get here
} catch (e) {
  // if we get here, we need to check for the expected error:
  assertEqual(e.errorNum, arangodb.ERROR_ARANGO_INDEX_CREATION_FAILED.code)
}
```